x-pack/plugins/security_solution/public/detection_engine/rule_creation/components/related_integrations/related_integration_field.tsx

```diff
@@ -0,0 +1,207 @@
+
+interface RelatedIntegrationItemFormProps {
+  field: FieldHook<RelatedIntegration>;
+  relatedIntegrations: RelatedIntegration[];
+  onRemove: () => void;
+}
+
+export function RelatedIntegrationField({
+  field,
+  relatedIntegrations,
+  onRemove,
+}: RelatedIntegrationItemFormProps): JSX.Element {
+  const { data: integrations, isInitialLoading } = useIntegrations();
+  const [integrationOptions, selectedIntegrationOptions] = useMemo(() => {
+    const currentKey = getKey(field.value.package, field.value.integration);
+    const relatedIntegrationsButCurrent = relatedIntegrations.filter(
+      (ri) => getKey(ri.package, ri.integration) !== currentKey
+    );
+    const unusedIntegrations = filterOutUsedIntegrations(
+      integrations ?? [],
+      relatedIntegrationsButCurrent
+    );
+
+    const options = unusedIntegrations.map(transformIntegrationToOption) ?? [];
+    const fallbackSelectedOption =
+      field.value.package.length > 0
+        ? {
+            key: currentKey,
+            label: `${capitalize(field.value.package)} ${field.value.integration ?? ''}`,
+          }
+        : undefined;
+    const selectedOption =
+      options.find((option) => option.key === currentKey) ?? fallbackSelectedOption;
+
+    return [options, selectedOption ? [selectedOption] : []];
+  }, [integrations, field.value, relatedIntegrations]);
+
+  const [packageErrorMessage, versionErrorMessage] = useMemo(() => {
+    const packagePath = `${field.path}.package`;
+    const versionPath = `${field.path}.version`;
+
+    return [
+      field.errors.find((err) => 'path' in err && err.path === packagePath),
+      field.errors.find((err) => 'path' in err && err.path === versionPath),
+    ];
+  }, [field.path, field.errors]);
+
+  const handleIntegrationChange = useCallback(
+    ([changedSelectedOption]: Array<EuiComboBoxOptionOption<Integration>>) =>
+      field.setValue({
+        package: changedSelectedOption?.value?.package_name ?? '',
+        integration: changedSelectedOption?.value?.integration_name,
+        version: changedSelectedOption?.value
+          ? calculateRelevantSemver(changedSelectedOption.value)
+          : '',
+      }),
+    [field]
+  );
+
+  const handleVersionChange = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) =>
+      field.setValue((oldValue) => ({
+        ...oldValue,
+        version: e.target.value,
+      })),
+    [field]
+  );
+
+  return (
+    <EuiPanel color="subdued">
+      <EuiFormRow
+        fullWidth
+        isInvalid={Boolean(packageErrorMessage) || Boolean(versionErrorMessage)}
+        error={packageErrorMessage?.message ?? versionErrorMessage?.message}
+      >
+        <EuiFlexGroup alignItems="center">
+          <EuiFlexItem grow={8}>
+            <EuiComboBox<Integration>
+              options={integrationOptions}
+              renderOption={renderIntegrationOption}
+              selectedOptions={selectedIntegrationOptions}
+              singleSelection
```

Comment:
Yes, I considered this option but favoured in to the first approach since it allows to choose different colors for the selected item to code installed and enabled states which conforms with the [design](https://github.com/elastic/kibana/issues/173595#issuecomment-2020039424). It's not visible on your screenshots so made two to compare

<img width="954" alt="Screenshot 2024-04-12 at 15 55 24" src="https://github.com/elastic/kibana/assets/3775283/103d1fdb-f2ca-44e6-a908-a45984af5677">

<img width="951" alt="Screenshot 2024-04-12 at 15 55 50" src="https://github.com/elastic/kibana/assets/3775283/1aed8e69-f62a-4538-957a-9db6f822ff18">

Reply:
Oh, I see you have removed the "delete cross" button within a pill. This looks 👍 to me now!